collator/src/mempool/impls/std_impl.rs

```diff
@@ -377,16 +393,41 @@ impl MempoolAdapter for MempoolAdapterStdImpl {
 
     async fn get_anchor_by_id(
         &self,
+        top_processed_to_anchor: MempoolAnchorId,
         anchor_id: MempoolAnchorId,
-    ) -> Result<Option<Arc<MempoolAnchor>>> {
-        Ok(self.cache.get_anchor_by_id(anchor_id).await)
+    ) -> Result<GetAnchorResult> {
+        // to be reproducible on all nodes regardless already committed data
+
+        let result = match self
+            .expect_running(top_processed_to_anchor, anchor_id)
+            .await
+        {
+            Some(true) => match self.cache.get_anchor_by_id(anchor_id).await {
+                Some(anchor) => GetAnchorResult::Exist(anchor),
+                None => GetAnchorResult::NotExist,
+            },
+            Some(false) | None => GetAnchorResult::MempoolPaused, // also paused until init
```

Comment:
have doubts on what to return on `None`. It's the case when `Engine` is not run yet , but anchors are already requested.

Reply:
How it worked before when the `expect_running()` check was not present?

Reply:
It returned `None` immediately, even in case when anchor could be received (or not) after some time. That behavior was inherited from old versions when mempool was unable to sync (regardless pause mode).

Reply:
it returned `NotExist` (`None`), but anchor may be received after mempool is run.